src/mlpack/core/math/random.hpp

```diff
@@ -23,12 +23,26 @@ namespace math /** Miscellaneous math routines. */ {
  * correctly on Windows.
  */
 
-// Global random object.
-extern MLPACK_EXPORT std::mt19937 randGen;
-// Global uniform distribution.
-extern MLPACK_EXPORT std::uniform_real_distribution<> randUniformDist;
-// Global normal distribution.
-extern MLPACK_EXPORT std::normal_distribution<> randNormalDist;
+//! Global random object.
+inline std::mt19937& RandGen()
+{
+  static std::mt19937 randGen;
+  return randGen;
+}
+
+//! Global uniform distribution.
+inline std::uniform_real_distribution<>& RandUniformDist()
+{
+  static std::uniform_real_distribution<> randUniformDist(0.0, 0.1);
```

Comment:
```suggestion
  static std::uniform_real_distribution<> randUniformDist(0.0, 1.0);
```
I suspect this might be the issue. :)